libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/extensions/MapboxFlowableExtensions.kt

```diff
@@ -97,3 +103,32 @@ fun MapboxNavigation.flowOnFinalDestinationArrival(): Flow<RouteProgress> = call
     registerArrivalObserver(observer)
     awaitClose { unregisterArrivalObserver(observer) }
 }
+
+@ExperimentalPreviewMapboxNavigationAPI
+fun MapboxNavigation.postCustomEvent(
+    payload: String,
+    @CustomEvent.Type customEventType: String,
+    navigationOptions: NavigationOptions,
+    customEventVersion: String,
+) {
```

Comment:
The receiver parameter (MapboxNavigation) is never used. You can correct that by removing the `navigationOptions` arg, and accessing it from the receiver.

```suggestion
fun MapboxNavigation.postCustomEvent(
    payload: String,
    @CustomEvent.Type customEventType: String,
    customEventVersion: String,
) {
```

Also, I think this extension function is in the wrong place. It's not a "flowable" extension.

Reply:
`navigationOptions` is being used in this extension.

```
runInTelemetryContext(navigationOptions) { telemetry ->
```

> Also, I think this extension function is in the wrong place. It's not a "flowable" extension.

Yeah I thought a lot about adding this here. I think we should revisit this file name. We call it `MapboxFlowableExtensions` but we use `MapboxNavigationEx` as JVM name. That's confusing to me because now I have to add a new `mapboxNavigation` extension and I am thinking how to name the JVM name for this new extension file. 🤔 why not just rename `MapboxFlowableExtensions` to `MapboxNavigationEx`?